xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-api/src/main/java/com/xwiki/macros/confluence/confluencepagetree/internal/macro/ConfluencePageTreeMacroParametersRefactoring.java

```diff
@@ -0,0 +1,150 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.macros.confluence.confluencepagetree.internal.macro;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.model.reference.SpaceReferenceResolver;
+import org.xwiki.rendering.block.MacroBlock;
+import org.xwiki.rendering.listener.reference.DocumentResourceReference;
+import org.xwiki.rendering.listener.reference.ResourceReference;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xwiki.macros.internal.updateReferences.AbstractReferenceUpdateMacroRefactoring;
+
+/**
+ * Implementation of reference refactoring operation for the confluence_pagetree macro.
+ *
+ * @version $Id$
+ * @since 1.29
+ */
+@Component
+@Named("confluence_pagetree")
+@Singleton
+public class ConfluencePageTreeMacroParametersRefactoring extends AbstractReferenceUpdateMacroRefactoring
+{
+    private static final String ROOT_PARAMETER = "root";
+
+    private static final String WEB_HOME = "WebHome";
+
+    @Inject
+    private DocumentReferenceResolver<String> documentReferenceResolver;
+
+    @Inject
+    private SpaceReferenceResolver<String> spaceReferenceResolver;
+
+    @Inject
+    @Named("compact")
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Inject
+    @Named("macro")
+    private EntityReferenceResolver<String> macroEntityReferenceResolver;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Override
+    public List<String> getParametersToUpdate()
+    {
+        return List.of(ROOT_PARAMETER);
+    }
+
+    @Override
+    public Set<ResourceReference> extractReferences(MacroBlock macroBlock)
+    {
+        String rootValue = macroBlock.getParameter(ROOT_PARAMETER);
+
+        XWikiContext xContext = contextProvider.get();
+        XWiki xWiki = xContext.getWiki();
+
+        try {
+            DocumentReference docRef = documentReferenceResolver.resolve(rootValue);
+
+            if (xWiki.exists(docRef, xContext)) {
```

Comment:
I don't think it's correct to check if the page exists here, this code should just return all possible references. What if the target page is created after this macro has been indexed?